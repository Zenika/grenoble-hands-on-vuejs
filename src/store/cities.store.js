import { defineStore } from 'pinia'
import { ref } from 'vue'

export const useCitiesStore = defineStore('cities', () => {
  // State
  const cities = ref([
    {
      name: 'Grenoble',
      latitude: 45.18924,
      longitude: 5.72213
    },
    {
      name: 'Singapour',
      latitude: 1.295600,
      longitude: 103.858995
    },
    {
      name: 'Bordeaux',
      latitude: 44.848089,
      longitude: -0.571017
    },
    {
      name: 'Brest',
      latitude: 48.389397,
      longitude: -4.499237
    },
    {
      name: 'MontrÃ©al',
      latitude: 45.523000,
      longitude: -73.581700
    },
    {
      name: 'Lyon',
      latitude: 45.767443,
      longitude: 4.858798
    },
    {
      name: 'Rennes',
      latitude: 48.113409,
      longitude: -1.661249
    },
    {
      name: 'Nantes',
      latitude: 47.207408,
      longitude: -1.556187
    },
    {
      name: 'Lille',
      latitude: 50.648670,
      longitude: 3.075520
    },
    {
      name: 'Paris',
      latitude: 48.878932,
      longitude: 2.328487
    },
  ])

  // Actions
  const getCityByName = cityName => cities.value.find(city => city.name === cityName)

  return { cities, getCityByName }
})
