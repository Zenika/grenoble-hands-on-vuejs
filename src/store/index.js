import { createStore } from 'vuex'

export default createStore({
  state: {
    latLongMap: {
      GRENOBLE: [45.183916, 5.703630],
      SINGAPOUR: [1.295600, 103.858995],
      BORDEAUX: [44.848089, -0.571017],
      BREST: [48.389397, -4.499237],
      MONTREAL: [45.523000, -73.581700],
      LYON: [45.767443, 4.858798],
      RENNES: [48.113409, -1.661249],
      NANTES: [47.207408, -1.556187],
      LILLE: [50.648670, 3.075520],
      PARIS: [48.878932, 2.328487]
    }
  },
  getters: {
    getMapPosition: state => state.latLongMap,
    getCityPosition: (state, getters) => cityName => getters.getMapPosition[cityName],
    getCities: state => Object.keys(state.latLongMap),
  },
  mutations: {
    createCity(state, city) {
      state.latLongMap[city.name] = [city.latitude, city.longitude]
    }
  },
  actions: {
    createCity(context, city) {
      context.commit('createCity', city)
    }
  },
  modules: {
  }
})
