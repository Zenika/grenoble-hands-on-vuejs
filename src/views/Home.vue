<template>
  <section>
    <h1 class="title">Offices</h1>

    <div class="panel">
      <router-link class="panel-block p-4" :to="{ name: 'City', params: { cityName: cities[0]}}">
        <h2 class="subtitle">{{ cities[0] }}</h2>
      </router-link>
    </div>
  </section>
</template>
<script>
export default {
  computed: {
    cities () {
      return this.$store.getters.getCities
    }
  }
}
</script>
