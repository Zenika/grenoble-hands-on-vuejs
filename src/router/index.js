import { createRouter, createWebHistory } from 'vue-router'
import Home from '@/views/Home.vue'
import City from '@/views/City.vue'

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/:cityName',
    name: 'City',
    props: true,
    component: City
  }
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
})

export default router
