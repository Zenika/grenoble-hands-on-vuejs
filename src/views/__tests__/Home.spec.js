import {shallowMount} from "@vue/test-utils";
import Home from "@/views/Home";
import router from "@/router";
import store from "@/store";

describe('Home component test all mocks', () => {
  let wrapper
  let $store
  beforeEach(() => {
    $store = {
      getters: {
        getCities: ['demo1', 'demo2']
      }
    }
    wrapper = shallowMount(Home, {
      global: {
        stubs: { 'router-link': true },
        mocks: {
          $store
        }
      }
    })
  });
  it('should render all cities in my world', () => {
    expect(wrapper.html()).toMatchSnapshot()
  });
});

describe('Home component test with instance', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Home, {
      global: {
        plugins: [store, router]
      }
    })
  });
  it('should render all cities where Zenika is', () => {
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('h1.title').text()).toBe('Offices')
    expect(wrapper.findAll('.panel-block.p-4').length).toBe(10)
  });
});
