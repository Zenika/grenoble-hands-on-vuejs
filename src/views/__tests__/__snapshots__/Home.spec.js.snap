// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test all mocks should render all cities in my world 1`] = `
<section>
  <h1 class="title">Offices</h1>
  <div class="panel">
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
  </div>
</section>
`;

exports[`Home component test with instance should render all cities where Zenika is 1`] = `
<section>
  <h1 class="title">Offices</h1>
  <div class="panel">
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
    <router-link-stub class="panel-block p-4" to="[object Object]"></router-link-stub>
  </div>
</section>
`;
